### PR TITLE
Longer3D: Set STEP_TIMER TEMP_TIMER & TIMER_SERIAL in variant

### PR DESCRIPTION
**Pull request by tpruvot**
SoftwareSerial default is TIM5 if not set, let TIM2 & TIM4 free for PWM pins

TIM1 remains free for now...

### CODE REVIEW

File: buildroot/share/PlatformIO/variants/MARLIN_F103VE_LONGER/variant.h [hunk 1]
```diff
@@ -117,13 +117,23 @@ extern "C" {
 #define PIN_WIRE_SCL            PB10
 
 // Timer Definitions
-// Use TIM6/TIM7 when possible as servo and tone don't need GPIO output pin
+// Let TIMER 2 for optional Fan PWM
+#ifndef TEMP_TIMER
+  #define TEMP_TIMER            3
+#endif
+// Let TIMER 4 for TFT backlight PWM or Servo freq...
```
Comment (thinkyhead) [inline buildroot/share/PlatformIO/variants/MARLIN_F103VE_LONGER/variant.h]: English tip: The word "Leave" is better than "Let" with this usage.

Comment (tpruvot) [reply buildroot/share/PlatformIO/variants/MARLIN_F103VE_LONGER/variant.h]: yep, or keep ;)

**Comment by tpruvot** [comment]
To note, hardware PWM on our board is new, we were unable to use correctly the pin alt functions with libmaple HAL... We tried for weeks with weird results and random crashes when trying to use afio_remap at runtime... Seems much more easy with the new one, pins are set correctly without explicit remap code... But i didn't try to create a variant for the F1 HAL, at least for the pins/timer array like we have now.

**Comment by thinkyhead** [comment]
> unable to use correctly the pin alt functions with libmaple

That's fine. We don't need every board to have a Maple build. The only potential advantage is that Maple may be better supported in Arduino IDE than stm32duino.

**Pull request merged**